src/Symfony/Component/Notifier/Bridge/Slack/SlackOptions.php

```diff
@@ -191,4 +198,11 @@ public function threadTs(string $threadTs): self
 
         return $this;
     }
+
+    private function validateNumberOfBlocks(): void
```

Comment:
IMHO no need for a new method for this. Especially because use case is different:
in constructor we prevent creating the object with a wrong payload (if exception is triggered, the object is not created)
In the `block` method, when the exception is triggered, it let the object in an invalid state, and we should avoid this.